ts/Series/Column/ColumnSeries.ts

```diff
@@ -636,9 +643,9 @@ class ColumnSeries extends Series {
                 // #3169, drilldown from null must have a position to work from.
                 // #6585, dataLabel should be placed on xAxis, not floating in
                 // the middle of the chart.
-                point.isNull ? translatedThreshold : barY,
-                barW,
-                point.isNull ? 0 : barH
+                point.isNull ? translatedThreshold - nullMinSpace : barY,
```

Comment:
The `barY` and `barH` variables are already corrected for `minPointLength`, which is the exact equivalent of `nullMinSpace` but for zero points. See if we can combine those, at least do the corrections in the same place.